### PR TITLE
change sex without disconnecting

### PR DESCRIPTION
**Pull request by mekolat**
Sex can now be sent the same way as classes (without kicking the player). No idea why this wasn't done before.. my guess is that perhaps Mana didn't have a packet to update char data.

- - -
## requires https://github.com/themanaworld/tmwa-server-data/pull/331

### CODE REVIEW

**Comment by MadCamel** [comment]
Doesn't look like you're taking sex-locked equipment into account. Should at least be de-equipping any sex-locked equipment from the player.. Though that can also be done with scripts, so de-equipping everything is probably safer..

I believe the kick was to ensure equipment stats and scripts were handled correctly. Some stat buffs/equipment effects would linger after unequip and recalculation.

**Comment by mekolat** [comment]
@MadCamel the unequip is handled [like it always has](https://github.com/mekolat/tmwa/blob/master/src/changesex/char.cpp#L1268)

**Comment by wushin** [comment]
I believe this got fixed when o11c fixed the race sprite actually. As thats when tmwa starts sending the proper packet for M+

File: src/map/pc.cpp [hunk 1]
```diff
@@ -3509,8 +3509,7 @@ int pc_setparam(dumb_ptr<map_session_data> sd, SP type, int val)
             }
             break;
         case SP::SEX:
-            // this is a really bad idea
-            sd->sex = static_cast<SEX>(val);
+            chrif_char_ask_name(AccountId(), sd->status_key.name, 5, HumanTimeDiff());
```
Comment (mekolat) [inline src/map/pc.cpp]: I wish there was a way to actually send the value of ```set Sex``` to the char server so that there could be more than 2 sexes

**Pull request merged**